homeassistant/components/sensor/vultr.py

```diff
@@ -0,0 +1,120 @@
+"""
+Support for monitoring the state of Vultr Subscriptions.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.vultr/
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, CONF_NAME)
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.components.vultr import (
+    CONF_SUBSCRIPTION, ATTR_CURRENT_BANDWIDTH_USED, ATTR_PENDING_CHARGES,
+    DATA_VULTR)
+
+DEFAULT_NAME = '{} {}'
+DEPENDENCIES = ['vultr']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Monitored conditions: name, units, icon
+MONITORED_CONDITIONS = {
+    ATTR_CURRENT_BANDWIDTH_USED: ['Current Bandwidth Used', 'GB',
+                                  'mdi:chart-histogram'],
+    ATTR_PENDING_CHARGES: ['Pending Charges', 'US$',
+                           'mdi:currency-usd']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SUBSCRIPTION): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(MONITORED_CONDITIONS)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Vultr subscription (server) sensor."""
+    vultr = hass.data.get(DATA_VULTR)
+    if not vultr:
+        return False
+
+    subscription = config.get(CONF_SUBSCRIPTION)
+    name = config.get(CONF_NAME, DEFAULT_NAME)
+    monitored_conditions = config.get(CONF_MONITORED_CONDITIONS,
+                                      MONITORED_CONDITIONS)
+
+    sensors = []
+
+    for condition in monitored_conditions:
+        if condition not in MONITORED_CONDITIONS:
```

Comment:
This will never happen as the schema validation only allows entries in `MONITORED_CONDITIONS`.